src/info/worker.ts

```diff
@@ -858,14 +870,79 @@ function changeSubSegment() {
   const maxMag = new Vector(constrainedMax, constrainedMax).mag();
 
   // if outside of the initial movement from the home position the force is too high, ignore it
-  if (!atHomePos() && forceMag >= maxMag)
-      force.set(0, 0);
-  else
-      force.set(fx, fy);
+  if (!atHomePos() && forceMag >= maxMag) {
+    force.set(0, 0);
+  }
+  else {
+    if (!atHomePos()) {
+
+      // this will break if we have less than 10 points in a subsegment
+      //console.log(finishTransition);
+      if (finishTransition) {
+        const w = 21;
+        const i = 6;
+
+        const x1 = (i / w) * fx;
```

Comment:
Instead of having so many x and y variables, may be its better to have an array of x,y and fEEPrev values.
And then you just loop and have something like 
xValues[i] = (xValues.length - i) * feePrev[i]
yValues[i] = (yValues.length - i) * feePrev[i]
Only consider this suggestion if you feel its easy to accomodate.If you feel like you need to change this at too many places, then please refrain from doing this now. It can be done post CSUN as it may introduce regression.
@rayanisran , @sriGanna

Reply:
I'll quickly fix this.

Reply:
Mentioned on Slack, but for posterity: even with an update this should go through another round of testing since we're dealing with potential uncertainties that might pop up when setting Haply forces, but I don't think this is feasible right now with how limited time there is. A lot of the code in this file needs refactoring, so that will be part of a separate work-item post CSUN as well.